src/run.ts

```diff
@@ -33,7 +34,20 @@ export default async function run(): Promise<void> {
         if (existsSync(artifact.path)) {
           return [[...withFiles, artifact], withoutFiles]
         } else {
-          return [withFiles, [...withoutFiles, artifact]]
+          const globsWithFiles = fastGlob.sync(artifact.path)
+          if (globsWithFiles.length !== 0) {
+            return [
+              withFiles.concat(
+                globsWithFiles.map(file => ({
+                  ...artifact,
+                  path: file
+                }))
+              ),
+              withoutFiles
+            ]
```

Comment:
One thing to note- we'll need to re-work some of the test results displays and flaky tests to merge artifacts within a single job which have the same name.

In other words, if I have 10 tests in file 1 and 15 in file 2, we'll probably just want to show "25 xUnit results" or whatever (that'll additionally be important for the flaky test detection code, which compares test results from artifacts across two jobs)

Reply:
We'll probably want to do the server-side work before we publish this to v1

Reply:
Could we provide both an aggregated view and an individual view? I could see some scenarios where people would want to aggregate the test results, and others where they'd want to keep them separated. I was going to suggest adding a configuration option like `aggregate: true|false`, but if the UI is built to easily view aggregated or disaggregated, then we don't need to configure anything on upload.

Reply:
For displaying in the UI, that would be possible (though we probably want to store the original file name, so it’s clear what’s from what. Right now we only have the name)

For flaky tests, I’d worry we’ll have misses if parallel distribution puts results in different files so I’d probably still prefer if those were all rolled up